pkg/controller/workload/job/job_controller.go

```diff
@@ -337,18 +328,22 @@ func (r *JobReconciler) ensureAtMostOneWorkload(ctx context.Context, job *batchv
 		}
 	}
 
-	// Delete redundant workload instances.
+	// Delete duplicate workload instances.
+	existedWls := 0
 	for i := range toDelete {
-		if err := r.client.Delete(ctx, toDelete[i]); err != nil {
+		err := r.client.Delete(ctx, toDelete[i])
+		if err == nil || !apierrors.IsNotFound(err) {
```

Comment:
IIUC，is  it  `if err == nil || apierrors.IsNotFound(err) {`

Reply:
This should be correct as is, what we want is to count the number of workloads that actually existed as of this moment. If the error is nil, we successful deleted the extra workload (so this workload existed and counts), if there was an error on deleting the workload, then we count only if the error was not NotFound.

Reply:
Thank you for the detailed explanation. 
A small suggestion: If we includes the `deleting error without NotFound` workload in `existedWls`, we maybe not say `deleted existedWls existing workload(s)` in error. Because some of them may be deleted failed and still existed. This can be confusing in the process of troubleshooting through the logs.

Can we change to `try to delete`or record success or failure separately ?

Reply:
changed to "tried deleting ..."